sorting/selection_sort_recursive.cpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file
+ * @brief Implementation of the [Selection
+ * sort](https://en.wikipedia.org/wiki/Selection_sort)
+ * implementation using recursion
+ * @details
+ * C++ program to sort an array using a recursive selection sort algorithm
```

Comment:
Please provide more details about the algorithm itself, how does it work, etc.. You can take the description of Wikipedia but adjust it to match our standards. 🙂